Critical revision of the CASTOR monitor package and migration to the new DQM interface + postprocessing

vanbesien opened the pull request:
This PR supersedes #7221 from Vladimir Popov.

We additionally:
- fixed a problem with deletion of CastorChannelQualityMonitor and CastorDataIntegrityMonitor
- closed a memoryleak in CastorMonitorModule

audrium committed:
added ZDCLowGainFraction condition object

audrium committed:
added Hcal prefix to the classes

audrium committed:
added Hcal prefix to the classes

audrium committed:
added record and plugin

audrium committed:
removed payload

audrium committed:
added record to HcalAllRcds.h

audrium committed:
added methods to CalibCalorimetry/HcalPlugins

audrium committed:
added new cond obj to CondTools and CalibCalorimetry packages

audrium committed:
small fixes

audrium committed:
changed ZDCLowGainFractions DataRecord

audrium committed:
fixed HcalDbASCIIIO.cc

Commit:
Removed asserts preventing fast re-recoes a la ECALELF

audrium committed:
changed cout to LogInfo -- HcalZDCLowGainFractionsHandler.cc

audrium committed:
fixed memory leak - deleted myDBObject variable in the end of the job -- HcalZDCLowGainFractionsPopConAnalyzer.cc

audrium committed:
added HcalTopology object into the new condition -- HcalZDCLowGainFractionsPopConAnalyzer.cc

jhgoh committed:
migration of legacy EDProducers to globa/stream modules

jhgoh committed:
override keyword

webermat committed:
update JetMET DQM to enable MiniAOD running, introduce PFCandidate maps for MET scanning

webermat committed:
rename MonitorElements, fix typo in JetCorrector call

matteosan1 committed:
add fillDescription

Commit:
getCollections

audrium committed:
removed setting of the HcalTopology object from the HcalZDCLowGainFractionsPopConAnalyzer.cc

audrium committed:
Merge pull request #2 from audrium/audrium/ZDCLowGainFraction_new_cond

removed setting of the HcalTopology object from the HcalZDCLowGainFracti...

popovvp committed:
package critical revision

popovvp committed:
package critical revision

popovvp committed:
package critical revision

popovvp committed:
package critical revision

popovvp committed:
package critical revision

popovvp committed:
package critical revision

popovvp committed:
package critical revision

popovvp committed:
package critical revision

popovvp committed:
package critical revision

popovvp committed:
package critical revision

Commit:
add counter

matteosan1 committed:
Merged topic_fillDescription from repository matteosan1

matteosan1 committed:
added fillDescription method

matteosan1 committed:
added more fillDescription

matteosan1 committed:
Merged refs/pull/7264/head from repository cms-sw

matteosan1 committed:
remove pippo

olivito committed:
update photon path names

matteosan1 committed:
updates

matteosan1 committed:
update ElectronSeedProducer

matteosan1 committed:
final update to ElectronSeedProducer

popovvp committed:
removed obsolete file

popovvp committed:
removed obsolete file

popovvp committed:
removed obsolete file

popovvp committed:
removed obsolete file

popovvp committed:
removed obsolete file

popovvp committed:
removed obsolete file

popovvp committed:
removed obsolete file

popovvp committed:
removed obsolete file

popovvp committed:
removed obsolete file

popovvp committed:
removed obsolete file

popovvp committed:
removed obsolete file

popovvp committed:
removed obsolete file

popovvp committed:
removed obsolete file

vanbesien committed:
Merged refs/pull/7221/head from repository cms-sw

vanbesien committed:
Really removed CastorChannelQualityMonitor and
CastorDataIntegrityMonitor

These files were intended to be removed, but somehow ended up being just
"empty". The files still existed, but had no content and size 0.

vanbesien committed:
Closed memoryleak in CastorMonitorModule

New deleting the different sub modules in the destructor.

silviodonato committed:
Merged refs/pull/7261/head from repository cms-sw

cmsbuild commented:
A new Pull Request was created by @vanbesien (Broen van Besien) for CMSSW_7_4_X.

Critical revision of the CASTOR monitor package and migration to the new DQM interface + postprocessing

It involves the following packages:

DQM/CastorMonitor


@cmsbuild, @danduggan, @nclopezo, @deguio can you please review it and eventually sign? Thanks.
You can sign-off by replying to this message having '+1' in the first line of your reply.
You can reject by replying  to this message having '-1' in the first line of your reply.
If you are a L2 or a release manager you can ask for tests by saying 'please test' in the first line of a comment.

degano commented:
@cmsbuild please test

Commit:
LL exo relvals

Commit:
LL exo relvals

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
+1
Tested at: 3e231b260cbc7a63d8b9687f92539ab1d70af829
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-7365/2119/summary.html

silviodonato committed:
../python/hltHiggsPostProcessors_cff.py

vanbesien committed:
Removing begin/endJob method +cleaning away done()

Removed beginJob and placed content in dqmbeginrun (since it needs
iSetup)
Removed endJob and also removed the corresponding done() methods in the
subclasses. Only one of these done() had content, which was stricly
verbose output.

vanbesien committed:
extra remove

cmsbuild commented:
Pull request #7365 was updated. @cmsbuild, @danduggan, @nclopezo, @deguio can you please check and sign again.

deguio commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.

deguio commented:
hello @popovvp 
we took the liberty to fix a couple of details in your PR. the behaviour should be unchanged.
waiting to see the outcome of the jenkins tests before approving.

cheers,
F.

silviodonato committed:
../python/hltHiggsPostProcessors_cff.py

silviodonato committed:
../python/hltHiggsValidator_cfi.py

silviodonato committed:
../python/hltHiggsValidator_cfi.py

Commit:
move photon tracking isolation collection to hiGeneralTracks

webermat committed:
code to run on MiniAOD and additional histograms for MET scanning and out of time pile up effects, reduce number of histograms we had so far

cmsbuild commented:
+1
Tested at: ec38fa9c4cdc462817dbcc10c18d5bae4e0d3adf
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-7365/2123/summary.html

istaslis committed:
hiDetachedTripletStep added

deguio commented:
I was checking the booking and I realized that, even if the number of histos is reduced wrt the previous version of the code, the memory explodes because of some 2D histograms like:
https://github.com/vanbesien/cmssw/blob/Castormonitor_postprocessing/DQM/CastorMonitor/src/CastorLEDMonitor.cc#L48
or all the histos at:
https://github.com/vanbesien/cmssw/blob/Castormonitor_postprocessing/DQM/CastorMonitor/src/CastorRecHitMonitor.cc#L35

1.12M of bins for a single 2D histo, is something we cannot afford. @popovvp please review the binning.

thanks,
F.

silviodonato committed:
hltHiggsValidator_cfi.py

silviodonato committed:
Merged VBFHbbDQM from repository silviodonato

istaslis committed:
Parameters updated

Commit:
EcalClusterLazyTools: added n5x5() in analogy with e5x5()

Commit:
check that rechits exist and satisfy quality criteria

makortel committed:
Ensure DetId belongs to Tracker before calling TrackerTopology::layer()

The rather simple fix escalated a bit to an attempt to make the code
more clear.

Commit:
Change the associator

Dr15Jones committed:
Removed unnecessary mutable keyword

The static analyzer complained about the existence of a non-thread
safe class being used as a mutable. Since the mutable keyword was
not necessary (the variable is never changed in a const member function)
it was removed.

ktf committed:
Get DataFormats/Math to compile on ICC.

wddgit committed:
Make OmniClusterRef data private

Dr15Jones committed:
Remove unused member variable from CastorDbService

An unused member variable was marked as mutable which was flagged
as a problem by the static analyzer.

smuzaffar committed:
removed unused include to avoid cyclic dependency

smuzaffar committed:
removed files which have been moved to RecoJets/JetProducers

friccita committed:
modeling pilot portcard electronics with new material

Dr15Jones committed:
Removed use of static from components method

The components method was using a static variable to hold the return
value from the function. Comments on the function stated it was hoped
that the routine would never be called. To fix the thread safety problem
and minimize the memory usage, the code was changed to use an std::atomic
pointer to cache the return value if and only if components was called.

This problem was found by the static analyzer.

webermat committed:
first version of code to run on MiniAOD, add PFCandidate maps and histograms for METScanning

ktf committed:
Workaround ICC error.

ktf committed:
Use clang fix also in the case of ICC.

perrotta committed:
Merged CMSSW_7_4_X from repository cms-sw

Dr15Jones committed:
Make GeometrySurface code fully const correct

Changed the return value of tangentPlane to be a 'const' smart pointer.
This avoids accidently changing internals data of the surface classes.

This problem was found by the static analyzer.

perrotta committed:
Integrate the HLT menu up to /dev/CMSSW_7_3_0/HLT/V97

Commit:
Merged HITrackingValidationWorkaround from repository kkrajczar

Commit:
remove unnecessary parameterchange

deguio committed:
update GT and connection string for condDB v2

deguio committed:
update GT and connection string for condDB v2

Commit:
update the PF hadron calibration code for 73X/74X

perrotta committed:
rebase

Commit:
Add an "--era" option to cmsDriver

ianna committed:
Update GT and expected result.

Commit:
Automatically add a bunchspacing era depending on the mix

ianna committed:
Merged validation-gt-update from repository ianna

Commit:
Add a customise function to perform the changes not yet implemented by the run2 era

Commit:
FastSim: a strange mechanism to place the primary sim vertex was removed. The primary sim vertex is now the first vertex in the hepmc event

friccita committed:
moved non-xml files to Geometry/TrackerCommonData/test directory

friccita committed:
removed deleted files

Commit:
Add additional online monitoring information to DQM^2.

Also fixes a bug which causes monitoring to update too fast.
And removes a few scripts (they live outside this repository
now).

Commit:
update the list of Hgg paths monitored

bendavid committed:
update to lhe interface for pythia 8.205

Commit:
Expand out every function call in customiseRun2EraExtras

bendavid committed:
remove TimeShower:nPartonsInBorn from FXFX merging fragment since it is no longer needed in pythia 8.205 when multiplicities are steered automatically

cmsbuild committed:
Merge pull request #7398 from Dr15Jones/removeUnneededMutableFromCastorDbService

Remove unused member variable from CastorDbService

Dr15Jones committed:
Initialize member std::atomic<...*> to nullptr

The C++ standard does not require that the default constructor of
std::atomic<T*> initialize the pointer to nullptr.

fwyzard committed:
hltIntegrationTests: increase the cmsRun stack size for multithreading

fwyzard committed:
hltIntegrationTests: increase the cmsRun stack size for multithreading

ktf committed:
Merge pull request #7396 from Dr15Jones/removeUnneededMutableFromPerformancePayloadFromTFormula

Removed unnecessary mutable keyword from PerformancePayloadFromTFormula

wddgit committed:
Bug fix in Halo code

Fix a bug where the code is using the index
into the RefVector where the index into the
collection is expected. This will result in
Ref's to the wrong element being created.

Dr15Jones committed:
Return a const MagVolume from findVolume

The static analyzer complained that the const function, findVolume,
was returning a non-const pointer to data being held by the object.
Although no code was found which altered the returned value, making
the return be const avoids possible problems in the future.

Dr15Jones committed:
Made Reflection::solid a const member function

The static analyzer complained that a class which held a Reflection
object by pointer was calling a non-const function, 'solid', of
Reflection from a const function of the holding class. Since 'solid'
never actually changed the state of a Reflection it was trivial
to change to be const.

wmtan committed:
Activate all products needed by Sim/Reco-level mixing

cmsbuild committed:
Merge pull request #7410 from perrotta/possiblyLastHltIntegrationsBeforeFreezing

Integrate HLT menu up to  /dev/CMSSW_7_3_0/HLT/V97

cmsbuild committed:
Merge pull request #7422 from dmitrijus/dqm_online_monitoring_jan28

Add additional online monitoring information to DQM^2.

cmsbuild committed:
Merge pull request #7432 from fwyzard/hltIntegrationTests_stack_size_73x

hltIntegrationTests: increase the cmsRun stack size for multithreading

cmsbuild committed:
Merge pull request #7431 from fwyzard/hltIntegrationTests_stack_size_74x

hltIntegrationTests: increase the cmsRun stack size for multithreading

cmsbuild committed:
Merge pull request #7426 from dildick/from-CMSSW_7_3_0_patch1-forDisplacedMuHLT

Add minN parameter to HLTMuonTrkFilter

cmsbuild committed:
Merge pull request #7436 from Dr15Jones/makeSolidConst

Made Reflection::solid a const member function

ktf committed:
Merge pull request #7406 from ktf/fix-intel2

Misc ICC errors fixes.

ktf committed:
Merge pull request #7395 from ktf/fix-icc

Get DataFormats/Math to compile on ICC.

davidlange6 committed:
Merge pull request #7400 from smuzaffar/from-CMSSW_7_4_X_2015-01-27-0200

removed unused include to avoid cyclic dependency

ktf committed:
Keep track of hostname being used to run jobs.

ktf committed:
Merge pull request #7440 from ktf/track-hostname

Keep track of hostname being used to run jobs.

cmsbuild committed:
Merge pull request #6615 from audrium/ZDCLowGainFraction_new_cond

new condition object (ZDC Low Gain Fraction) added for HCAL

emanueledimarco committed:
integrate the fast multifit for HLT in 740pre5, yet not using the DB inputs

ianna committed:
Cleanup unneeded classValidation/Geometry/BuildFile.xml

threus committed:
added plot with cluster multiplicity vs BX and time

cmsbuild committed:
Merge pull request #7438 from wmtan/FixSimRecoMixingIn7_4_X

Activate all products needed by Sim/Reco-level mixing (in 7_4_X)

davidlange6 committed:
Merge pull request #7447 from bendavid/lheupdate_py8205_74x

Lhe interface update for pythia 8.205 (74x)

cmsbuild committed:
Merge pull request #7419 from mark-grimes/convertPostLS1ToEraCore

Implement postLS1 as an Era - core parts

Commit:
adapt workflow name

Commit:
adapt workflow name

Commit:
adapt workflow name

Commit:
adapt workflow name

Commit:
RecoHI/HiJetAlgos: resolve GCC warnings about unknown pragma

GCC correctly ignores unknown Clang pragma. These warnings needs to be
disabled in command line and not via #pragma GCC diagnostic, which does
not correctly with GCC (a long standing bug in C++ preprocessor).

Marking #pragma as GCC will not help as warning is Clang specific, not
yet supported in GCC.

Signed-off-by: David Abdurachmanov <David.Abdurachmanov@cern.ch>

jalimena committed:
add customization for hltL2CosmicMuonSeeds for muon NoBPTX paths

Commit:
fix bug in step name

Commit:
fix bug in step name

Commit:
another fix for bug in step name

Commit:
another fix for bug in step name

cmsbuild committed:
Merge pull request #7441 from emanueledimarco/ecalmultifit-hlt-from740pre5

integrate the fast multifit for HLT in 740pre5, yet not using the DB inp...

ptraczyk committed:
cleanup muon timing

ptraczyk committed:
switch off ECAL in combined muon timing

Martin-Grunewald committed:
Update to slove mereg conflicts

wddgit committed:
Bug fix in HLTJetCollectionsVBFFilter

Fix a bug where the code is using the index
into the RefVector where the index into the
collection is expected. This will result in
Ref's to the wrong element being created.

cmsbuild committed:
Merge pull request #7452 from jalimena/hltL2CosmicMuonSeedsCustomization

customization for hltL2CosmicMuonSeeds

bachtis committed:
Changing also cluster HO Threshold

cmsbuild committed:
Merge pull request #6625 from shervin86/GsfElectronFix

Removed asserts preventing fast re-recoes a la ECALELF

cmsbuild committed:
Merge pull request #7374 from richard-cms/hiPhotonIsolationFix

HeavyIons: switch photon tracking isolation collection to hiGeneralTracks

mkirsano committed:
remove obsolete code

cmsbuild committed:
Merge pull request #7403 from Dr15Jones/fixedThreadSafetyInRecoTrackerTkDetLayers

Removed use of static from components method

mkirsano committed:
clean test directory

cmsbuild committed:
Merge pull request #7402 from smuzaffar/remove-migrated-files

removed files which have been moved to RecoJets/JetProducers

mkirsano committed:
remove extensions

Dr15Jones committed:
Remove improper call to edm::Ref constructor call

The associator was constructing an edm::Ref by passing it an
edm::RefVector and an index. However, the index used was one into
the edm::RefVector but the constructor requires the index to be back
to the original collection to which the edm::RefVector associates.
The only way the associator ever worked correctly was when the edm::RefVector
used contained entries for every item in the original collection.
The problematic edm::Ref constructor will be removed in another commit.
This problem was found by David Dagenhart.

cmsbuild committed:
Merge pull request #7397 from wddgit/omniClusterRef

Make OmniClusterRef data private

ktf committed:
Merge pull request #7409 from Dr15Jones/constCorrectGeometry

Make GeometrySurface code fully const correct

ktf committed:
Merge pull request #7435 from Dr15Jones/constCorrectnessForMagField

Return a const MagVolume from findVolume

cmsbuild committed:
Merge pull request #7462 from mkirsano/cleanthepeg

Cleanthepeg

cmsbuild committed:
Merge pull request #7461 from lveldere/fastsim-fix-primary-simvertex

fix to primary vertex in fastsim simevent

webermat committed:
first version of code to run on MiniAOD, add PFCandidate maps and hisâ¦ â¦

pastika committed:
Updates to HCAL relval macros and histogram binning

Updated relval code including replacing some example config and macro files in CaloTowers/test.
Updated histogram range for 2 histogrms in HcalRecHits/src/HcalRecHitsValidation.cc

cmsbuild committed:
Merge pull request #7384 from jhgoh/MultithreadedMuonHLT

multithreading migration for HLT muon

cmsbuild committed:
Merge pull request #7450 from davidlt/resolve-clang-pragma-warning-in-GCC

RecoHI/HiJetAlgos: resolve GCC warnings about unknown pragma

cmsbuild committed:
Merge pull request #7449 from shervin86/LazyTools

Retrieve collections from LazyTools

cmsbuild committed:
Merge pull request #7424 from mmarionncern/PFCalib-73X

update the PF hadron calibration code for 73X/74X

cmsbuild committed:
Merge pull request #7457 from wddgit/bugMakingRefFromRefVector2

Bug fix in HLTJetCollectionsVBFFilter

Commit:
float => int in matrixSize()

cmsbuild committed:
Merge pull request #7455 from Martin-Grunewald/EcalMultiFitHLT

Migration to prepare for EcalMultiFit and GRun.txt fix

perrotta committed:
Merged refs/pull/7455/head from repository cms-sw

mandrenguyen committed:
Update hiDetachedTripletStep_cff.py

perrotta committed:
Fix: replace L1_EGgamma35 seeds when needed

perrotta committed:
Merged CMSSW_7_4_X from repository cms-sw

cmsbuild committed:
Merge pull request #7445 from makortel/phase1_fwd_TrackerTopology_fix

Fix in TrackerTopology migration of TrackingTruthAccumulator

cmsbuild committed:
Merge pull request #7458 from bachtis/HO74X_ClusterThresholds

Lowering of HO cluster Threshold in Particle Flow

rovere committed:
Use getByToken in CosmicTrackFinder

rovere committed:
Use getByToken in CosmicSeedGenerator

cmsbuild committed:
Merge pull request #7456 from mark-grimes/convertPostLS1ToEraTranche1

Implement postLS1 as an Era - tranche 1

cmsbuild committed:
Merge pull request #7467 from cms-tsg-storm/fixForL1EGamma35Seeds

Fix for L1_EGamma35 seeds

davidlange6 committed:
Merge pull request #7264 from matteosan1/topic_fillDescription

Add fillDescription methods.

davidlange6 committed:
Merge pull request #7463 from Dr15Jones/fixBadRefConstructorCall

Remove improper call to edm::Ref constructor call

cmsbuild committed:
Merge pull request #7465 from pastika/7_4_X_HCALRelval_Update_1836_29_1_2015

Updates to HCAL relval macros and histogram binning

cmsbuild committed:
Merge pull request #7448 from threus/clusmulti_vs_bx_orbit

added plot with cluster multiplicity vs BX and time

cmsbuild committed:
Merge pull request #7443 from ianna/validation-geometry-cleanup

Validation Geometry Cleanup

cmsbuild committed:
Merge pull request #7420 from friccita/PilotElectronicsMaterial

Pilot Electronics Material

deguio commented:
-1
rejecting for now. it's on @popovvp to take actions.
F.

cmsbuild committed:
Merge pull request #7414 from deguio/updateToCondDBv2_74x

update GT and connection string for condDB v2

cmsbuild committed:
Merge pull request #7297 from olivito/smpdqm_photon_74x

update photon path names for SMP HLT DQM

cmsbuild committed:
Merge pull request #7404 from kkrajczar/HITrackingValidationWorkaround

Change the track associator in HI Track Validation

cmsbuild committed:
Merge pull request #6859 from JetMETdqmval/develop_DQM_74X_V0

update JetMET DQM to enable MiniAOD running, introduce PFCandidate maps ...

cmsbuild committed:
Merge pull request #7423 from HuguesBrun/updateHiggsDQMin740pre5LowMassHiggs

update the list of Hgg paths monitored

cmsbuild committed:
Merge pull request #7387 from silviodonato/VBFHbbDQM

DQM for HLT VBF Hbb

cmsbuild committed:
Merge pull request #7388 from konush/LazyTools_74X_n5x5

EcalClusterLazyTools: new n5x5() call analogous to e5x5()

bmarzocc committed:
Run2_AlcaPhiStream

bmarzocc committed:
Run2_AlcaPhiStream

cmsbuild committed:
Merge pull request #7433 from wddgit/bugMakingRefFromRefVector

Bug fix in Halo code

cmsbuild committed:
Merge pull request #7454 from ptraczyk/cleanupMuonTiming

Cleanup muon timing

rovere committed:
Use getByToken in CosmicTrackingRegion

bmarzocc committed:
remove redundant descriptions line

rovere committed:
Use getByToken in CRackSeedGenerator

Commit:
modified path names

perrotta committed:
Last cleanup before freezing the menu

Commit:
shorten wmlhe GEN workflow names; affects only injection, and no running / IB

cmsbuild committed:
Merge pull request #7366 from srimanob/llexorelval

LL exo relvals

cmsbuild committed:
Merge pull request #7478 from cms-tsg-storm/fixForL1EGamma35Seeds

Last cleanup before freezing the menu

cmsbuild committed:
Merge pull request #7481 from franzoni/shorten-GEN-wfName-CMSSW_7_4_0_pre6

Shorten wmlhe wf name

cmsbuild committed:
Merge pull request #7474 from rovere/fixGetByLabels

Fix get by labels

cmsbuild committed:
Merge pull request #7473 from ndaci/ExoValDev_74Xto73X_PR3

Update on muon NoBPTX path names

cmsbuild committed:
Merge pull request #7393 from CmsHI/addHiDetachedTripletStep

Add hi detached triplet step

ktf committed:
Merge pull request #7487 from gartung/clangSA-update-class-list

Clang sa update class list

cmsbuild committed:
Merge pull request #7471 from bmarzocc/Run2_AlcaPhiStream

Run2 alca phi stream

vanbesien committed:
Merged refs/pull/7365/head from repository cms-sw

vanbesien committed:
Commented away 4 very large historgrams

vanbesien commented:
I'm closing this one.
Created #7499 instead.
